test/distributed/test_c10d_nccl.py

```diff
@@ -282,66 +315,50 @@ def allreduce(tensors, op):
             work.wait()
 
         # Sum
-        tensors = []
-        for i in range(self.num_gpus):
-            tensors.append(torch.tensor([i + 1]).cuda(i))
+        tensors = [torch.tensor([self.rank + 1]).cuda(local_device_id)]
 
         allreduce(tensors, c10d.ReduceOp.SUM)
 
-        for i in range(self.num_gpus):
-            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-            self.assertEqualIgnoreType(
-                torch.tensor([float(self.num_gpus * (self.num_gpus + 1) / 2)]),
-                tensors[i],
-            )
+        # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
+        self.assertEqualIgnoreType(
+            torch.tensor([float(self.world_size * (self.world_size + 1) / 2)]),
+            tensors[0],
+        )
 
         # Avg (only available for NCCL 2.10+)
         if torch.cuda.nccl.version() >= (2, 10, 0):
-            tensors = [torch.tensor([i + 1.]).cuda(i) for i in range(self.num_gpus)]
+            tensors = [torch.tensor([self.rank + 1.]).cuda(local_device_id)]
 
             allreduce(tensors, c10d.ReduceOp.AVG)
 
-            for i in range(self.num_gpus):
-                # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
-                ndev = float(self.num_gpus)
-                self.assertEqualIgnoreType(
-                    torch.tensor([ndev * (ndev + 1.) / (2. * ndev)]),
-                    tensors[i],
-                )
+            # TODO(#38095): Replace assertEqualIgnoreType. See issue #38095
+            ndev = float(self.world_size)
```

Comment:
Nit: we can move it before previous `assertEqualIgnoreType` so that this can be reused.